token/supr.go

```diff
@@ -0,0 +1,162 @@
+	Classification1   string            `json:"classification1"`
+	Classification2   string            `json:"classification2"`
+	Classification3   string            `json:"classification3"`
+	ManagedInSupr     bool              `json:"managed_in_supr"`
+	APIOpaqueData     string            `json:"api_opaque_data"`
+	NgiSensitiveData  bool              `json:"ngi_sensitive_data"`
+	NgiReady          bool              `json:"ngi_ready"`
+	NgiDeliveryStatus string            `json:"ngi_delivery_status"`
+	ContinuationName  string            `json:"continuation_name"`
+	StartDate         string            `json:"start_date"`
+	EndDate           string            `json:"end_date"`
+	Pi                Pi                `json:"pi"`
+	Members           []Member          `json:"members"`
+	LinksOutgoing     []interface{}     `json:"links_outgoing"`
+	LinksIncoming     []interface{}     `json:"links_incoming"`
+	Resourceprojects  []ResourceProject `json:"resourceprojects"`
+	Modified          string            `json:"modified"`
+}
+
+type Pi struct {
+	ID        int    `json:"id"`
+	FirstName string `json:"first_name"`
+	LastName  string `json:"last_name"`
+	Email     string `json:"email"`
+}
+
+type Member struct {
+	ID        int    `json:"id"`
+	FirstName string `json:"first_name"`
+	LastName  string `json:"last_name"`
+	Email     string `json:"email"`
+}
+
+type ResourceProject struct {
+	ID                   int          `json:"id"`
+	Allocated            int          `json:"allocated"`
+	Resource             Resource     `json:"resource"`
+	DecommissioningState string       `json:"decommissioning_state"`
+	Allocations          []Allocation `json:"allocations"`
+}
+
+type SuprHeader struct {
+	APIVersion       string `json:"apiVersion"`
+	Code             int    `json:"code"`
+	Service          string `json:"service"`
+	DeveloperMessage string `json:"developerMessage"`
+	UserMessage      string `json:"userMessage"`
+	ErrorCode        int    `json:"errorCode"`
+	DocLink          string `json:"docLink"`
+}
+
+type SuprUserResult struct {
+	Username     string `json:"username"`
+	SSHPublicKey string `json:"sshPublicKey"`
+	PasswordHash string `json:"passwordHash"`
+	UID          int    `json:"uid"`
+	Gecos        string `json:"gecos"`
+}
+
+type Resource struct {
+	ID            int    `json:"id"`
+	Name          string `json:"name"`
+	CapacityUnit  string `json:"capacity_unit"`
+	CapacityUnit2 string `json:"capacity_unit_2"`
+	Centre        Center `json:"centre"`
+}
+
+type Allocation struct {
+	ID        int    `json:"id"`
+	StartDate string `json:"start_date"`
+	EndDate   string `json:"end_date"`
+	Allocated int    `json:"allocated"`
+}
+
+type Center struct {
+	ID   int    `json:"id"`
+	Name string `json:"name"`
+}
+
+// verifyProjectAccount checks that the given `email` is actually
+// the PI of the given `project_id` and returns error otherwise
+func verifyProjectAccount(username string, projectID string) error {
```

Comment:
It seems there's quite some code repetition with `verifyEGABoxAccount` but this can wait for a refactoring sprint :)